pkg/virt-handler/migration-proxy/migration-proxy.go

```diff
@@ -366,6 +378,29 @@ func (m *migrationProxy) StopListening() {
 	}
 }
 
+func formatAddress(targetAddress string) string {
+	// In case address is already formatted ([ipv6]:port), no need to do anything
+	if strings.Contains(targetAddress, "[") {
+		return targetAddress
+	}
+
+	// Unix socket doesnt need to be formatted
+	if strings.Count(targetAddress, ":") == 0 {
+		return targetAddress
+	}
+
+	stringSlice := strings.Split(targetAddress, ":")
```

Comment:
If you get an ipv6 socket with no brackets it means the sender of this socket has a bug. IMO, you shouldn't fix a non-valid ipv6 socket here, the sender should send the correct format. If you get a wrong format from the sender you should fail.

Reply:
it means we need to find all the situations of the callers, and for each one of them wrap the ip,
also need to take into account that if there are cases where the migration that i will run doesnt pass across , i wont be able to find them so fast

Reply:
@oshoval could you rephrase? I can't follow the reply.

Reply:
I don't understand why we expect situations where we don't get either a valid ipv4 or ipv6 address.

Reply:
Hi, sure

Generally saying that if we want that the caller will have the responsibility to format its ip:port as [ipv6]:port, in case needed (ipv6 and not ipv4 / unix socket),
it means that i will need to find all the places where it call the function, and add there the logic that checks if wrap is needed / wrap according. 

just wonder what is the right approach you both prefer, since Miguel and i liked more the approach that already exists, at least from what i understood on previous comments, but you both decide.

Thanks

Reply:
> I don't understand why we expect situations where we don't get either a valid ipv4 or ipv6 address.

callers doesnt format the ipv6:port as [ipv6]:port
so according this comment, i believe you prefer the way Alona said as well,
to make the caller format the TargetAddress

Reply:
Not sure I can follow. If something creates invalid sockets, then that should be fixed. Does that answer the question? We should not create wrong things on one end, only to fix it then somewhere else.

Reply:
You can always guard against invalid input and fail with noise, but trying to fix input somehow is not good.

Reply:
it doesn't create wrong sockets since this trap and format it well, but i understand that you prefer that its will be the caller responsibility to format it well and not the endpoint, as Alona asked, thanks on it.